homeassistant/components/plugwise/climate.py

```diff
@@ -0,0 +1,250 @@
+        """Return current active hvac state."""
+        if self._api.get_schema_state(self._domain_objects) is True:
+            return HVAC_MODE_AUTO
+        return HVAC_MODE_OFF
+
+    @property
+    def preset_mode(self):
+        """Return the active preset mode."""
+        return self._api.get_current_preset(self._domain_objects)
+
+    @property
+    def preset_modes(self):
+        """Return the available preset modes list without values."""
+        presets = list(self._api.get_presets(self._domain_objects).keys())
+        return presets
+
+    @property
+    def hvac_modes(self):
+        """Return the available hvac  modes list."""
+        return self._hvac_modes
+
+    @property
+    def current_temperature(self):
+        """Return the current temperature."""
+        return self._api.get_temperature(self._domain_objects)
+
+    @property
+    def min_temp(self):
+        """Return the minimal temperature possible to set."""
+        return self._min_temp
+
+    @property
+    def max_temp(self):
+        """Return the maximum temperature possible to set."""
+        return self._max_temp
+
+    @property
+    def target_temperature(self):
+        """Return the target temperature."""
+        return self._api.get_target_temperature(self._domain_objects)
+
+    @property
+    def temperature_unit(self):
+        """Return the unit of measured temperature."""
+        return TEMP_CELSIUS
+
+    def set_temperature(self, **kwargs):
+        """Set new target temperature."""
+        _LOGGER.debug("Adjusting temperature")
+        temperature = kwargs.get(ATTR_TEMPERATURE)
+        if temperature is not None and self._min_temp < temperature < self._max_temp:
+            self._temperature = temperature
+            self._api.set_temperature(self._domain_objects, temperature)
+            _LOGGER.debug("Changing temporary temperature")
+        else:
+            _LOGGER.error("Invalid temperature requested")
+
+    def set_hvac_mode(self, hvac_mode):
+        """Set the hvac mode."""
+        _LOGGER.debug("Adjusting hvac_mode (i.e. schedule/schema)")
+        if self._previous_schema is None and self._active_schema is None:
+            _LOGGER.error("previous_schema not known, unable to apply")
+        else:
+            if self._previous_schema is None:
+                schema = self._active_schema
+            else:
+                schema = self._previous_schema
+            schema_mode = "false"
+            if hvac_mode == HVAC_MODE_AUTO:
+                schema_mode = "true"
+            self._api.set_schema_state(self._domain_objects, schema, schema_mode)
+
+    def set_preset_mode(self, preset_mode):
+        """Set the preset mode."""
+        _LOGGER.debug("Adjusting preset_mode (i.e. preset)")
+        if preset_mode in self._preset_modes:
+            self._preset_mode = preset_mode
+            self._api.set_preset(self._domain_objects, preset_mode)
+            _LOGGER.debug("Changing preset mode")
```

Comment:
I would put the logging before calling the api, since we don't log the result of the api call.

Reply:
Thanks!